std/equal.m31

```diff
@@ -375,6 +375,16 @@ let process_eta =
     | |- ?e : ?t => deep [] e t
     end
 
+let process_reducing c l =
+  let _ = match c with
+      | |- _constant _ => ()
+      | |- _atom _ => ()
+      | |- _proj _ _ => ()
+      | _ => print "only constants and atoms may have reducing information"; failure
```

Comment:
Well then, we should/could better than that. If you think about how the code is written, it's better to have `failure XXX` than `print XXX ; failure`. What would be needed? Maybe a builtin operation that fails gracefully?